ignitionrobotics/ign-gazebo
# Consolidate scene3D with ign-gui

Signed-off-by: ahcorde <ahcorde@gmail.com>
# 🎉 New feature

This PR is part of the consolidation of Scene3D with ign-gazebo's GzScene3D https://github.com/ignitionrobotics/ign-gui/issues/137

Related to:
 - https://github.com/ignitionrobotics/ign-rendering/pull/311
 - https://github.com/ignitionrobotics/ign-gui/pull/217

## Summary

This new plugin makes use of the Scene3DInterface provided in the ign-gui PR

## Test it

replace the GZScene3D with this:

```xml
<plugin filename="Scene3DNew" name="3D View">
  <ignition-gui>
    <title>3D View</title>
    <property type="bool" key="showTitleBar">false</property>
    <property type="string" key="state">docked</property>
  </ignition-gui>

  <engine>ogre2</engine>
  <scene>scene</scene>
  <ambient_light>0.4 0.4 0.4</ambient_light>
  <background_color>0.8 0.8 0.8</background_color>
  <camera_pose>6 0 6 0 0.5 3.14</camera_pose>
  <fullscreen>0</fullscreen>
</plugin>
```

and run 

```bash
ign gazebo -r -v 4 diff_drive.sdf --gui-config gui.config
```

You will see:

 - The scene with the model
 - sky enabled
 - try moveto
 - try follow

## Checklist

**Note to maintainers**: Remember to use **Squash-Merge**
